base/log_level.go

```diff
@@ -0,0 +1,60 @@
+package base
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+)
+
+// LogLevel is used to represent a log level.
+type LogLevel uint32
+
+const (
+	// LEVEL_NONE disables all logging
+	LEVEL_NONE LogLevel = iota
+	// LEVEL_ERROR enables only error logging.
+	LEVEL_ERROR
+	// LEVEL_WARN enables warn, and error logging.
+	LEVEL_WARN
+	// LEVEL_INFO enables info, warn, and error logging.
+	LEVEL_INFO
+	// LEVEL_DEBUG enables all logging.
+	LEVEL_DEBUG
+)
+
+var logLevelNames = []string{"none", "error", "warn", "info", "debug"}
```

Comment:
This feels a bit brittle - particularly with things like LogLevelName() having the potential to panic if this gets out of step with the LogLevel enum above.  This could get protected w/ unit tests, but I don't think it's covered currently.  (e.g. adding an entry to the enum could cause a panic in LogLevelName without TestLogLevelNames failing).  Maybe just modify TestLogLevelNames to iterate over all LogLevel values?

Reply:
Good catch, thanks. I've added an assertion in TestLogLevelNames to ensure the length of this slice, and the number of constants are equal, and now return an empty string for out of bounds lookups.